src/ducks/files/rename.js

```diff
@@ -0,0 +1,73 @@
+/* global cozy */
+
+import { getFiles } from './files'
+import { META_DEFAULTS as meta, extractFileAttributes } from '../../actions'
+
+// constants
+
+const START_RENAMING = 'START_RENAMING'
+const ABORT_RENAMING = 'ABORT_RENAMING'
+export const RENAME_SUCCESS = 'RENAME_SUCCESS'
+const UPDATE_FILE_NAME = 'UPDATE_FILE_NAME'
+const RENAME_FAILURE_DUPLICATE = 'RENAME_FAILURE_DUPLICATE'
+
+// reducers
+
+const initialState = { file: null, name: null }
+const renameReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case START_RENAMING:
+      return { ...state, file: action.file, name: action.file.name }
+    case UPDATE_FILE_NAME:
+      return { ...state, name: action.name }
+    case RENAME_SUCCESS:
+    case ABORT_RENAMING:
+    case RENAME_FAILURE_DUPLICATE:
+      return initialState
+    default:
+      return state
+  }
+}
+export default renameReducer
+
+// selectors
+
+export const isRenaming = state => state.rename !== initialState
+export const getRenamingFile = state => state.rename.file
+export const getUpdatedName = state => state.rename.name
```

Comment:
The selectors of a sub redux state should correspond to that specific sub redux state as mentioned at 1'05 in the [Redux: Colocating Selectors with Reducers](https://egghead.io/lessons/javascript-redux-colocating-selectors-with-reducers) egghead video.